File: SoftUni-Python-Fundamentals/p04_functions/exercises/password_validator.py

def valid_password(password):
    if 6 <= len(password) <= 10 and string.isalnum() and digits >= 2:
        print("Password is valid")
    if not 6 <= len(password) <= 10:
        print("Password must be between 6 and 10 characters")
    if not password.isalnum():
        print("Password must consist only of letters and digits")
    if not digits >= 2:
        print("Password must have at least 2 digits")

string = str(input())
digits = 0
for x in range(len(string)):
    if string[x].isdigit():
        digits += 1
valid_password(string)

File: SoftUni-JS-Front-End/functionsAndStatementsExercise/passwordValidator.js
function solve(password) {
    let regexAlNum = /^[a-zA-Z0-9]+$/
    let array = password.split("")
    let incorrectPassword = false
    let numberCount = 0
    if ((password.length < 6)||(password.length > 10)) {
        incorrectPassword = true
        console.log("Password must be between 6 and 10 characters")
    }
    if (!regexAlNum.test(password)) {
        incorrectPassword = true
        console.log("Password must consist only of letters and digits")
    }
    for (let i of array) {
        if (!isNaN(i)) {
            numberCount += 1
        }
    }
    if (numberCount <= 1) {
        incorrectPassword = true
        console.log("Password must have at least 2 digits")
    }
    if (!incorrectPassword) {
        console.log("Password is valid")
    }
}

solve('My12Pass')
